channeldb/invoice_test.go

```diff
@@ -330,29 +293,24 @@ func TestInvoiceAddTimeSeries(t *testing.T) {
 		}
 	}
 
+	var settledInvoices []Invoice
+
 	// We'll now only settle the latter half of each of those invoices.
 	for i := 10; i < len(invoices); i++ {
 		invoice := &invoices[i]
 
 		paymentHash := invoice.Terms.PaymentPreimage.Hash()
 
-		_, err := db.UpdateInvoice(
+		updatedInvoice, err := db.UpdateInvoice(
 			paymentHash, getUpdateInvoice(0),
 		)
 		if err != nil {
 			t.Fatalf("unable to settle invoice: %v", err)
 		}
-	}
 
-	invoices, err = db.FetchAllInvoices(false)
-	if err != nil {
-		t.Fatalf("unable to fetch invoices: %v", err)
+		settledInvoices = append(settledInvoices, *updatedInvoice)
```

Comment:
Doesn't this reduce the effect of this test, because we rely on the code under test to provides us with the expectations?

Reply:
I'd argue that it actually makes the test more strict as before we relied on `FetchAllInvoices` too. Ideally expected results should be constant and possbily included in the test itself rather then generated/selected etc.

Reply:
The removal of `FetchAllInvoices` is good, but it is now replaced by something that uses the code under test to generate the expectations. This test is testing whether the filtering is correct. I'd say we don't need to do the deep equal on the exact invoice contents for that. It is tested elsewhere already. For me, building a list of payment hashes here and asserting on it in the various filter cases is enough.

Reply:
I'm not convinced that this is not a test relaxation (compared to what it was before). Maybe it's more correct this way though. PTAL.

Reply:
Isn't the `updatedInvoice` still used as an expectation now?

Reply:
Oh, it is changed in a later commit. Generally we don't want to touch the same code in one pr twice. Switching them around as you had before lowers the summed delta of all commits.

Reply:
Sorry, that was a rebase mistake on my end. Corrected.